pkg/flow/internal/controller/loader.go

```diff
@@ -75,6 +75,15 @@ func (l *Loader) Apply(parentScope *vm.Scope, blocks []*ast.BlockStmt) diag.Diag
 		newGraph dag.Graph
 	)
 
+	// Pre-populate graph with a ConfigNode.
+	// TODO (@tpaschalis) Are we okay with recreating the config node on each
+	// Apply? Should we check if it already exists and just update its blocks
+	// instead? if exist := l.graph.GetByID(configNodeID); exist == nil { ...
```

Comment:
Yeah, this is probably fine for now. The main reason we don't want to recreate other nodes is because they manage an instance of a component. Recreating vs updating here doesn't seem to have a similar impact that recreating a component node does.

Reply:
Ok, makes sense.